pkg/controller/render/render_controller.go

```diff
@@ -636,3 +763,447 @@ func getMachineConfigsForPool(pool *mcfgv1.MachineConfigPool, configs []*mcfgv1.
 	}
 	return out, nil
 }
+
+func (ctrl *Controller) experimentalAddImageStreams(pool *mcfgv1.MachineConfigPool) error {
+
+	var err error
+
+	// Make sure we have an image stream to render content into
+	renderedImageStreamName := pool.Name + ctrlcommon.ImageStreamSuffixRenderedConfig
+
+	// And also, an imagestream to receive our "coreos-derive" build image
+	contentImageStreamName := pool.Name + ctrlcommon.ImageStreamSuffixMcoContent
+
+	// And allow for the user to specify some stuff afterwards
+	customImageStreamName := pool.Name + ctrlcommon.ImageStreamSuffixMcoContentCustom
+
+	// And then if we're feeling dangerous, allow for an external image
+	externalImageStreamName := pool.Name + ctrlcommon.ImageStreamSuffixMcoContentExternal
+
+	// Our list of imagestreams we need to ensure exists
+	var ensureImageStreams = []string{
+		renderedImageStreamName,
+		contentImageStreamName,
+		customImageStreamName,
+		externalImageStreamName,
+		"coreos",
+	}
+
+	glog.V(2).Infof("Ensuring image streams exist for pool %s", pool.Name)
+
+	// Make sure the imagestreams exist so we can populate them with our image builds
+	for _, imageStreamName := range ensureImageStreams {
+
+		_, err = ctrl.imageclient.ImageV1().ImageStreams(ctrlcommon.MCONamespace).Get(context.TODO(), imageStreamName, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			// If we don't, we need to make it, otherwise our builds will fail
+
+			newImageStream := &imagev1.ImageStream{}
+			newImageStream.Name = imageStreamName
+			newImageStream.Namespace = ctrlcommon.MCONamespace
+			newImageStream.Spec.LookupPolicy.Local = false
+
+			// Set ownerships so these get cleaned up if we delete the pool
+			// TODO(jkyros): I have no idea if this actually cleans the images out of the stream if we delete it?
+			poolKind := mcfgv1.SchemeGroupVersion.WithKind("MachineConfigPool")
+			oref := metav1.NewControllerRef(pool, poolKind)
+			newImageStream.SetOwnerReferences([]metav1.OwnerReference{*oref})
+
+			// coreos imagestream is base, it's special, it needs to pull that image
+			if imageStreamName == "coreos" {
+
+				controllerConfig, err := ctrl.ccLister.Get(ctrlcommon.ControllerConfigName)
+				if err != nil {
+					return fmt.Errorf("could not get ControllerConfig %v", err)
+				}
+
+				newImageStream.Spec = imagev1.ImageStreamSpec{
+					LookupPolicy:          imagev1.ImageLookupPolicy{Local: false},
+					DockerImageRepository: "",
+					Tags: []imagev1.TagReference{
+						{
+							Name: "latest",
+							From: &corev1.ObjectReference{
+								Kind: "DockerImage",
+								Name: controllerConfig.Spec.BaseOperatingSystemContainer,
+							},
+						},
+					},
+				}
+
+			}
+
+			// It didn't exist, put the imagestream in the cluster
+			_, err := ctrl.imageclient.ImageV1().ImageStreams(ctrlcommon.MCONamespace).Create(context.TODO(), newImageStream, metav1.CreateOptions{})
+			if err != nil {
+				return fmt.Errorf("Attempted to create imagestream %s but failed: ", err)
```

Comment:
Formatting looks wrong, this should include `imageStreamName` and also use `%w` to pass the error.

Reply:
I did fix this, thank you.